content/articles/isp-management-system/index.md

```diff
@@ -0,0 +1,786 @@
+3. Basic knowledge of MongoDB.
+4. You should also have [Node](https://nodejs.org/en/download/) installed on your machine.
+
+### Table of Contents
+
+- [Backend](#backend)
+- [Frontend](#frontend)
+- [Connecting Frontend to Backend](#Connecting-Frontend-to-Backend)
+- [Conclusion](#conclusion)
+
+### Backend
+The web application will follow the architecture shown below  consisting of the Client, Server, and Database.
+
+
+We will handle all the backend functionality of our app using a [REST API](https://www.smashingmagazine.com/2018/01/understanding-using-rest-api/) created using Node, Express, and MongoDB.
+- **MongoDB:** A database that uses a document-oriented data model.
+- **ExpressJS:** A web application framework for building web applications and APIs.
+- **Node.js:** A cross-platform JavaScript runtime environment used to build servers, and applications.
+
+#### Setting up Development Environment
+
+Let us first set up our development environment.
+Create a new folder and navigate to the new folder by running the following commands on a terminal
+
+``mkdir api``
+``cd api``
+
+Inside this folder, run `npm init--y` to get a starter package.json which will contain all the dependencies we will need to run the app.
+Our package.json file gets updated when we install various dependencies.
+Then install Mongoose,Express, and MongoDB using the following command.
+``npm install express mongodb mongoose --save``
+
+We will use express to set up a node serve.
+MongoDB will enable us to connect to the database, and mongoose, which we will use for defining models.
+
+We will be sharing resources from the server to the client; therefore, we install cors to handle cross-origin resource sharing.
+
+To install cors run this command on the terminal.
+``npm install cors``
+
+We use the cors package to enable the server to shares its resources only with clients on a different domain.
+You can learn more about CORS [here](https://en.wikipedia.org/wiki/Cross-origin_resource_sharing).
+
+We will install nodemon to run our server automatically when we make changes.
+
+``npm install nodemon --save-dev``
+
+Change the scripts in package.json to 
+
+``"devStart": "nodemon index.js"``
+
+We will also install dotenv to handle our variables from a .env file.
+
+``npm install dotenv --save-dev``
+
+> nodemon and dotenv are in our development dependencies.
+
+#### Setting up the Server
+
+Create a ``.env`` file within the api folder.
+We will add the database url which we will connect to and the port we will be listening for new connections.
+Add the following to the ``.env`` file.
+```
+DATABASE_URL = mongodb://localhost/customers
+PORT = 5000
+```
+
+Create ``index.js`` where we will set up the server.
+- Load the environment variables from the .env file by requiring and configuring the dotenv package.
+The environment variables are loaded into the ``process.env`` object. 
+Import the mongoose package and load the PORT variable.
+
```

Comment:
Every sentence is in a new line. Combine sentences to a paragraph. Ensure it isn't too big.